### PR TITLE
ECC-674 Registered Address lookup

### PR DESCRIPTION
**Pull request by vigneshwaripandi**

### CODE REVIEW

File: app/uk/gov/hmrc/xieoricommoncomponentfrontend/controllers/SicCodeController.scala [hunk 1]
```diff
@@ -51,15 +58,31 @@ class SicCodeController @Inject() (
...
-                Redirect(uk.gov.hmrc.xieoricommoncomponentfrontend.controllers.routes.HavePBEController.onPageLoad())
-              )
+              groupEnrolment.getEori(loggedInUser).flatMap {
+                case Some(gbEori) =>
+                  subscriptionDisplayService.getSubscriptionAddress(gbEori).map {
+                    case Right(response) =>
+                      if (
+                        response.CDSEstablishmentAddress.postalCode.isDefined &&
+                        !response.CDSEstablishmentAddress.postalCode.isEmpty &&
+                        !response.CDSEstablishmentAddress.postalCode.get.startsWith("BT")
```
Comment (NandanaRaghu) [inline app/uk/gov/hmrc/xieoricommoncomponentfrontend/controllers/SicCodeController.scala:68]: Can we define the check in private method?

File: app/uk/gov/hmrc/xieoricommoncomponentfrontend/services/SubscriptionDisplayService.scala [hunk 2]
```diff
@@ -56,4 +56,23 @@ class SubscriptionDisplayService @Inject() (
...
+          case Right(resp) =>
+            sessionCache.saveSubscriptionDisplay(resp).map(
+              successfulWrite =>
+                if (!successfulWrite)
+                  logger.debug("SubscriptionDisplay SUB09 details retrieved successfully and saved into cache")
+            )
+          case Left(_) => logger.debug("SubscriptionDisplay SUB09 call failed and details are not saved into cache")
+        }
+        subscriptionDisplayResponse
+    }
```
[the remark this answers is not in the document]
Comment (vigneshwaripandi) [reply app/uk/gov/hmrc/xieoricommoncomponentfrontend/services/SubscriptionDisplayService.scala]: I have changed the code to reuse the existing method

File: app/uk/gov/hmrc/xieoricommoncomponentfrontend/controllers/SicCodeController.scala (hunk 1 above)
Comment (vigneshwaripandi) [reply app/uk/gov/hmrc/xieoricommoncomponentfrontend/controllers/SicCodeController.scala]: I have added the new method as requested. Please re-review.

**Review by NandanaRaghu — APPROVED** [review]
LGTM

**Pull request merged**